src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -149,16 +200,36 @@ private string GetAssemblyReferencesCommands()
             return result.ToString();
         }
 
+        protected override string GenerateCommandLineCommands()
+        {
+            if (UseCrossgen2 && !string.IsNullOrEmpty(DotNetHostPath))
```

Comment:
Should we also check `&& !IsPdbCompilation`? BTW, it might be cleaner to have a flag like this:
```C#
private bool ActuallyUseCrossgen2 => UseCrossgen2 && !IsPdbCompilation;
```

Reply:
Added in 6th commit.